api/views/user_views.py

```diff
@@ -714,3 +714,14 @@ def get(self, request, recording_id):
 
         serializer = QuizSerializer(quizzes, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class TokenVerificationView(APIView):
+    permission_classes = [AllowAny]
+
+    def get(self, request, token):
+        if not Token.objects.filter(key=token).exists():
+            return JsonResponse(
+                {"message": "Token is invalid"}, status=status.HTTP_401_UNAUTHORIZED
+            )
+        return JsonResponse({"message": "Token is valid"}, status=status.HTTP_200_OK)
```

Comment:
Let's simplify the logic to something like this:
```
class TokenVerificationView(APIView):
    permission_classes = [IsAuthenticated]

    def get(self, request):
        return Response({"message": "Token is valid"}, status=status.HTTP_200_OK)
```

We should also add a `throttle_classes` to this endpoint.